internal/resources/grafana/resource_sso_settings.go

```diff
@@ -271,6 +280,12 @@ func ReadSSOSettings(ctx context.Context, d *schema.ResourceData, meta interface
 	}
 
 	settingsSnake := make(map[string]any)
+
+	customFieldsFromTfState := make(map[string]any)
```

Comment:
Are the field not all returned because they're sensitive?

Reply:
All custom fields should be returned by the API, but the concern here is that custom fields are not separated from the main fields in the API response. Also the API might return other fields that we want to ignore and not consider them custom fields by default. That's why I wanted to check the existing custom fields from the terraform state and match them against the ones retrieved from the API.

Reply:
hmm ok gotcha. So only what's been written by TF is managed. Are there specific fields that we want to ignore. As it is, the read functionality is not great for a TF resource

Reply:
No, unfortunately there aren't any specific fields we want to ignore from the API because the API considers the settings fields as an opaque `map[string]any`.